### ISSUE TITLE
Populate local simulators configuration and properties

### ISSUE THREAD

**Issue by diego-plan9** [Qiskit/qiskit-terra#1332]
### What is the current behavior?

Currently, after #1323 (and unmerged #1331), the local simulators have a number of placeholder values for some fields of those structures:

* `configuration.gates`
```
'gates': [{'name': 'TODO', 'parameters': [], 'qasm_def': 'TODO'}]
```
* `properties.qubits`
* `properties.gates`
```
            'qubits': [[{'name': 'TODO', 'date': '2000-01-01 00:00:00Z',
                         'unit': 'TODO', 'value': 0}]],
            'gates': [{'qubits': [0], 'gate': 'TODO',
                       'parameters':
                           [{'name': 'TODO', 'date': '2000-01-01 00:00:00Z',
                             'unit': 'TODO', 'value': 0}]}],
            'general': []
```
* `properties.general` ? (can be empty).

Those are for conformance with the specification - they require that those lists have at least one item. We should populate them properly (or alternatively, decide if the schemas need to be relaxed for the simulators).

### Suggested solutions

Some fields might be able to be generated using the existing information (such as `basis_gates`) and the terra infrastructure - an approach might be to include some tooling to generate them, instead of copy-pasting.

**Comment by diego-plan9** [comment]
An update: @Max-Reuter is working on the configuration, and I'll be working on not having properties for the simulators.

**Comment by diego-plan9** [comment]
After #1401 and #1413, the only thing left would be to update the `qasm_def` of some gates (the ones that are not part of the standard library, which still contain some placeholder values). Considering the upcoming moving of the simulators, I'm closing the issue - @jaygambetta please reopen if this is something that should be tackled pre-move to aer.
